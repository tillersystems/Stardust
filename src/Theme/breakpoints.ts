/**
 * Defines application breakpoints.
 */

const xs = 0;
const sm = 576;
const md = 768;
const lg = 992;
const xl = 1200;

export const breakpoints = {
  xs,
  sm,
  md,
  lg,
  xl,
};
