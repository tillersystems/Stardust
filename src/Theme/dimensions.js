/**
 * Defines the theming dimensions.
 */

const radius = 0.4;
const tiny = 0.3;
const small = 0.7;
const medium = 1.5;
const big = 3.1;
const huge = 4.5;

export const dimensions = {
  radius: `${radius}rem`,

  tiny: `${tiny}rem`,
  small: `${small}rem`,
  medium: `${medium}rem`,
  big: `${big}rem`,
  huge: `${huge}rem`,

  radiusInt: radius,

  tinyInt: tiny,
  smallInt: small,
  mediumInt: medium,
  bigInt: big,
  hugeInt: huge,
};
