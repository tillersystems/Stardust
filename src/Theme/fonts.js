/**
 * Defines the theming of fonts.
 */

export const fonts = {
  size: {
    // -start- Old font size
    tiny: '0.3rem',
    small: '0.7rem',
    // medium: '1.2rem',
    // big: '3.2rem',
    huge: '6.4rem',
    // -end- Old font size

    // -start- New font size
    default: '1.2rem',
    medium: '1.4rem',
    big: '1.6rem',
    h6: '1.8rem',
    h5: '2.0rem',
    h4: '2.2rem',
    h3: '2.6rem',
    h2: '3rem',
    h1: '3.6rem',
    // -end- New font size

    // -start- Old font int
    tinyInt: 0.3,
    smallInt: 0.7,
    // mediumInt: 1.2,
    // bigInt: 3.2,
    hugeInt: 6.4,
    // -end- Old font int

    // -start- Old font int
    defaultInt: 1.2,
    mediumInt: 1.4,
    bigInt: 1.6,
    h6Int: 1.8,
    h5Int: 2.0,
    h4Int: 2.2,
    h3Int: 2.6,
    h2Int: 3,
    h1Int: 3.6,
    // -end- Old font int
  },

  spacing: {
    small: '0.5px',
    medium: '1px',
    big: '1.5px',
  },

  weight: {
    thin: 200,
    normal: 400,
    thick: 600,
  },

  family: '"Open Sans", Helvetica, Arial, sans-serif',
};
