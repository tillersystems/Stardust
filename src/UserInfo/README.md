# User Info

### Usage

```jsx
import { UserInfo } from '@tillersystems/stardust';
```

<!-- STORY -->

### Properties

- `className` - className needed by styled component.
- `name` - user name.
- `onClick` - onClick trigger.
- `pictureSrc` - source image of a user.

| propName     | propType | defaultValue | isRequired |
| ------------ | :------: | :----------: | :--------: |
| `className`  | `string` |     `''`     |     -      |
| `name`       | `string` |              |     \*     |
| `onClick`    |  `func`  |  `() => {}`  |     -      |
| `pictureSrc` | `string` |    `null`    |     -      |

### Example

```jsx
import { UserInfo } from '@tillersystems/stardust';

render() {
  return <UserInfo name={name} pictureSrc="http://examplePictureUrl.com" onClick={() => {}} />;
}
```
