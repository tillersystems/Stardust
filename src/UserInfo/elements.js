import styled from 'styled-components';
import breakpoint from 'styled-components-breakpoint';

export const UserName = styled.span`
  flex: 1;

  font-family: ${({ theme: { fonts } }) => fonts.family};
  font-weight: ${({ theme: { fonts } }) => fonts.weight.thin};
  letter-spacing: ${({ theme: { fonts } }) => fonts.spacing.medium};
  text-align: left;

  /* For medium devices */
  ${breakpoint('sm', 'lg')`
    display: none;
  `};
`;

export const DisconnectWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
