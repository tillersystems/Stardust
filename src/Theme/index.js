/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
