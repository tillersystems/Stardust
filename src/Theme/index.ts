import { DefaultTheme } from 'styled-components';

/**
 * Theme exported to the provider
 * This is globally available in styled-components when interpolating a function like so:
 * ${(props) => props.theme}
 * Or using import { withTheme } from 'styled-components';
 */

import { breakpoints } from './breakpoints';
import { palette } from './palette';
import { dimensions } from './dimensions';
import { fonts } from './fonts';

const Theme: DefaultTheme = {
  breakpoints,
  palette,
  dimensions,
  fonts,
};

export default Theme;
