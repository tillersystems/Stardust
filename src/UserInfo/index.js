import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import breakpoint from 'styled-components-breakpoint';

import { Avatar, Icon } from '..';
import { UserName, DisconnectWrapper } from './elements';

/**
 * User Info
 *
 * This component is in charge of displaying
 * the info of a current user
 *
 * @return {jsx}
 */

const UserInfo = ({ className, name, onClick, pictureSrc }) => (
  <div className={className}>
    <Avatar name={name} src={pictureSrc} />
    <UserName>{name}</UserName>
    <DisconnectWrapper onClick={onClick} data-test="DisconnectWrapper-test">
      <Icon color="white" name="power-off" />
    </DisconnectWrapper>
  </div>
);

/**
 * PropTypes Validation.
 */
const { func, string } = PropTypes;
UserInfo.propTypes = {
  className: string,
  name: string.isRequired,
  onClick: func,
  pictureSrc: string,
};

/**
 * Default props.
 */
UserInfo.defaultProps = {
  className: '',
  onClick: () => {},
  pictureSrc: null,
};

export default styled(UserInfo)`
  display: grid;
  grid-template-columns: 4.1rem auto auto;
  align-items: center;

  padding: ${({ theme: { dimensions } }) => dimensions.medium};

  background: rgba(0, 0, 0, 0.2);
  border-top: 1px solid ${({ theme: { palette } }) => palette.darkBlue};
  border-bottom: 1px solid ${({ theme: { palette } }) => palette.darkBlue};

  color: ${({ theme: { palette } }) => palette.white};
  font-size: ${({ theme: { fonts } }) => fonts.size.medium};

  cursor: pointer;

  /* For small devices */
  ${breakpoint('xs', 'sm')`
    display: none;
  `};

  /* For medium devices */
  ${breakpoint('sm', 'lg')`
    align-items: center;
    justify-content: center;
    grid-template-columns: auto;
    grid-template-rows: 4.1rem 5rem;
  `};
`;
