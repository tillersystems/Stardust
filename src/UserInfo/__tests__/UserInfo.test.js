import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import UserInfo from '..';
import Theme from '../../Theme';

describe('<UserInfo />', () => {
  it('should render withouth a problem', () => {
    const render = mount(
      <ThemeProvider theme={Theme}>
        <UserInfo name="Tony Starck" />
      </ThemeProvider>,
    );
    expect(render).toMatchSnapshot();
  });

  it('should render a user picture', () => {
    const render = mount(
      <ThemeProvider theme={Theme}>
        <UserInfo name="Tony Starck" pictureSrc="http://pictureSrc.com" />
      </ThemeProvider>,
    );
    expect(render).toMatchSnapshot();
  });

  it('should trigger onClick function when user click', () => {
    const spy = jest.fn();
    const render = mount(
      <ThemeProvider theme={Theme}>
        <UserInfo name="Tony Starck" pictureSrc="http://pictureSrc.com" onClick={spy} />
      </ThemeProvider>,
    );

    // console.log(render.debug());
    render
      .find('[data-test="DisconnectWrapper-test"]')
      .at(0)
      .simulate('click');

    expect(spy.mock.calls.length).toBe(1);
  });
});
