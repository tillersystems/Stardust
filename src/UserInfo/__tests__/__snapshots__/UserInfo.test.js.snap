// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> should render a user picture 1`] = `
.c2 {
  border-radius: 50%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 3.1rem;
  height: 3.1rem;
  border-radius: 50%;
  background: hsl(145,63%,49%);
  color: white;
}

.c5 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: "Open Sans",Helvetica,Arial,sans-serif;
  font-weight: 200;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-align: left;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: grid;
  grid-template-columns: 4.1rem auto auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.5rem;
  background: rgba(0,0,0,0.2);
  border-top: 1px solid hsl(213,17%,20%);
  border-bottom: 1px solid hsl(213,17%,20%);
  color: hsl(0,100%,100%);
  font-size: 1.4rem;
  cursor: pointer;
}

@media (min-width:36em) and (max-width:61.9375em) {
  .c3 {
    display: none;
  }
}

@media (min-width:0em) and (max-width:35.9375em) {
  .c0 {
    display: none;
  }
}

@media (min-width:36em) and (max-width:61.9375em) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    grid-template-columns: auto;
    grid-template-rows: 4.1rem 5rem;
  }
}

<Styled(UserInfo)
  name="Tony Starck"
  pictureSrc="http://pictureSrc.com"
>
  <UserInfo
    className="c0"
    name="Tony Starck"
    onClick={[Function]}
    pictureSrc="http://pictureSrc.com"
  >
    <div
      className="c0"
    >
      <Styled(Avatar)
        name="Tony Starck"
        src="http://pictureSrc.com"
      >
        <Avatar
          className="c1"
          colors="hsl(145, 63%, 49%)"
          name="Tony Starck"
          size={3.1}
          src="http://pictureSrc.com"
        >
          <div
            aria-label="Tony Starck"
            className="c1"
          >
            <Image
              alt="Tony Starck"
              height={31}
              src="http://pictureSrc.com"
              width={31}
            >
              <img
                alt="Tony Starck"
                className="c2"
                height={31}
                src="http://pictureSrc.com"
                width={31}
              />
            </Image>
          </div>
        </Avatar>
      </Styled(Avatar)>
      <styled.span>
        <span
          className="c3"
        >
          Tony Starck
        </span>
      </styled.span>
      <styled.div
        data-test="DisconnectWrapper-test"
        onClick={[Function]}
      >
        <div
          className="c4"
          data-test="DisconnectWrapper-test"
          onClick={[Function]}
        >
          <Styled(Icon)
            color="white"
            name="power-off"
          >
            <Icon
              className=""
              color="white"
              height="20"
              name="power-off"
              spin={false}
              width="20"
            >
              <styled.div
                height="20"
                spin={false}
                width="20"
              >
                <div
                  className="c5"
                  height="20"
                  width="20"
                >
                  <svg
                    height="20"
                    style={
                      Object {
                        "display": "block",
                      }
                    }
                    viewBox="0 0 512 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M376.7 156.736c-8.33-8.333-8.328-21.84.004-30.17 8.333-8.33 21.84-8.328 30.17.005 83.292 83.319 83.277 218.381-.033 301.682-83.31 83.301-218.372 83.301-301.682 0-83.31-83.3-83.325-218.363-.033-301.682 8.33-8.333 21.837-8.335 30.17-.005 8.332 8.33 8.335 21.837.005 30.17-66.634 66.655-66.622 174.705.026 241.346 66.648 66.64 174.698 66.64 241.346 0s66.66-174.691.026-241.346zM234.772 42.666c0-11.781 9.552-21.333 21.334-21.333s21.333 9.552 21.333 21.334V256c0 11.782-9.551 21.333-21.333 21.333-11.782 0-21.334-9.55-21.334-21.333V42.667z"
                      fill="hsl(0, 100%, 100%)"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </styled.div>
            </Icon>
          </Styled(Icon)>
        </div>
      </styled.div>
    </div>
  </UserInfo>
</Styled(UserInfo)>
`;

exports[`<UserInfo /> should render withouth a problem 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 3.1rem;
  height: 3.1rem;
  border-radius: 50%;
  background: hsl(145,63%,49%);
  color: white;
}

.c4 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: "Open Sans",Helvetica,Arial,sans-serif;
  font-weight: 200;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-align: left;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: grid;
  grid-template-columns: 4.1rem auto auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.5rem;
  background: rgba(0,0,0,0.2);
  border-top: 1px solid hsl(213,17%,20%);
  border-bottom: 1px solid hsl(213,17%,20%);
  color: hsl(0,100%,100%);
  font-size: 1.4rem;
  cursor: pointer;
}

@media (min-width:36em) and (max-width:61.9375em) {
  .c2 {
    display: none;
  }
}

@media (min-width:0em) and (max-width:35.9375em) {
  .c0 {
    display: none;
  }
}

@media (min-width:36em) and (max-width:61.9375em) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    grid-template-columns: auto;
    grid-template-rows: 4.1rem 5rem;
  }
}

<Styled(UserInfo)
  name="Tony Starck"
>
  <UserInfo
    className="c0"
    name="Tony Starck"
    onClick={[Function]}
    pictureSrc={null}
  >
    <div
      className="c0"
    >
      <Styled(Avatar)
        name="Tony Starck"
        src={null}
      >
        <Avatar
          className="c1"
          colors="hsl(145, 63%, 49%)"
          name="Tony Starck"
          size={3.1}
          src={null}
        >
          <div
            aria-label="Tony Starck"
            className="c1"
          >
            <span>
              TS
            </span>
          </div>
        </Avatar>
      </Styled(Avatar)>
      <styled.span>
        <span
          className="c2"
        >
          Tony Starck
        </span>
      </styled.span>
      <styled.div
        data-test="DisconnectWrapper-test"
        onClick={[Function]}
      >
        <div
          className="c3"
          data-test="DisconnectWrapper-test"
          onClick={[Function]}
        >
          <Styled(Icon)
            color="white"
            name="power-off"
          >
            <Icon
              className=""
              color="white"
              height="20"
              name="power-off"
              spin={false}
              width="20"
            >
              <styled.div
                height="20"
                spin={false}
                width="20"
              >
                <div
                  className="c4"
                  height="20"
                  width="20"
                >
                  <svg
                    height="20"
                    style={
                      Object {
                        "display": "block",
                      }
                    }
                    viewBox="0 0 512 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M376.7 156.736c-8.33-8.333-8.328-21.84.004-30.17 8.333-8.33 21.84-8.328 30.17.005 83.292 83.319 83.277 218.381-.033 301.682-83.31 83.301-218.372 83.301-301.682 0-83.31-83.3-83.325-218.363-.033-301.682 8.33-8.333 21.837-8.335 30.17-.005 8.332 8.33 8.335 21.837.005 30.17-66.634 66.655-66.622 174.705.026 241.346 66.648 66.64 174.698 66.64 241.346 0s66.66-174.691.026-241.346zM234.772 42.666c0-11.781 9.552-21.333 21.334-21.333s21.333 9.552 21.333 21.334V256c0 11.782-9.551 21.333-21.333 21.333-11.782 0-21.334-9.55-21.334-21.333V42.667z"
                      fill="hsl(0, 100%, 100%)"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </styled.div>
            </Icon>
          </Styled(Icon)>
        </div>
      </styled.div>
    </div>
  </UserInfo>
</Styled(UserInfo)>
`;
